smarts/core/coordinates.py

```diff
@@ -208,6 +228,20 @@ def __eq__(self, other) -> bool:
     def __hash__(self):
         return hash((*self.position, *self.orientation))
 
+    def reset_with(self, position, heading: Heading):
+        if self.position.dtype is not np.dtype(np.float64):
+            # The slice assignment below doesn't change self.position's dtype,
+            # which can be a problem if it was initialized with ints and
+            # now we are assigning it floats, so we just cast it...
+            self.position = np.float64(self.position)
+        self.position[:] = position
+        if "point" in self.__dict__:
+            # clear the cached_property
+            del self.__dict__["point"]
+        if heading != self.heading_:
+            self.orientation = fast_quaternion_from_angle(heading)
+            self.heading_ = heading
```

Comment:
Is the postfix `_` a convention? I am not quite familiar with it.

Reply:
I've seen it for some codebases, but it's less common.
I don't know why it was used here (`self.heading_` was like this from long ago, I believe -- I definitely didn't add it myself).